### PR TITLE
Fix `--accept` flag in `cabal-testsuite`

### PR DESCRIPTION
**Pull request by 9999years**
This was broken in #10225.

### CODE REVIEW

**Review by geekosaur — APPROVED** [review]

**Comment by ffaf1** [comment]
This is missing the PR template.

File: cabal-testsuite/src/Test/Cabal/Monad.hs [hunk 1]
```diff
@@ -880,10 +880,10 @@ testUserCabalConfigFile env = testCabalDir env </> "config"
 
 -- | The file where the expected output of the test lives
 --
--- Pointing to the @testTmpDir@ allows us to modify the expected output if
--- needed, to adapt it to outcomes of previous steps in the test.
+-- Note: This needs to point to `testSourceDir` so we write @--accept@ output
```
Comment (ffaf1) [inline cabal-testsuite/src/Test/Cabal/Monad.hs:883]:
```suggestion
-- Note: This needs to point to @testSourceDir@ so we write @--accept@ output
```

\` in Haddock is an hyperlinked identifier.

**Review by ffaf1 — APPROVED** [review]

File: cabal-testsuite/src/Test/Cabal/Monad.hs (hunk 1 above)
Comment (ffaf1) [reply cabal-testsuite/src/Test/Cabal/Monad.hs:883]: my bad

**Comment by jasagredo** [comment]
Ah I think I know why this is not failing anymore, its because of the tests marked as flaky, those have to modify the output in-place with previous steps output.

I think the right thing to do would be to export both expected output files, the one in the working directory and the one in the source, instead of unifying them.

**Comment by 9999years** [comment]
@jasagredo I've added back `testExpectFile` with the previous definition, but it's not used anywhere outside of `Test.Cabal.Monad`, so I'm not sure what you mean by flaky tests relying on it. (Which tests? What's a previous step in this context?)

**Comment by jasagredo** [comment]
I mean that some tests have to adapt the expected output in the middle of the execution, like [this one](https://github.com/haskell/cabal/blob/master/cabal-testsuite/PackageTests/NewUpdate/RejectFutureIndexStates/cabal.test.hs#L15) because precisely that outcome is what is being tested.

When checking the outcome, it will compare it with the locally modified version of `cabal.out`, which is undoable if we only refer to files in the source, not in the working dir.

However I wonder how this would play with `--accept` in this particular case...

**Pull request merged**